src/ninja.cc

```diff
@@ -710,26 +743,51 @@ int NinjaMain::ToolCompilationDatabase(const Options* options, int argc,
 
   optind = 1;
   int opt;
-  while ((opt = getopt(argc, argv, const_cast<char*>("hx"))) != -1) {
+  string err;
+  Node* user_given_target = NULL;
```

Comment:
Please create a scope in for `case 'a'` and move those two variable declarations there.

See http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-introduce